### PR TITLE
#165 - UMA : add creation and expiration resource support to oxd

### PR DESCRIPTION
**Pull request by duttarnab**
#165 - UMA : add creation and expiration resource support to oxd
https://github.com/GluuFederation/oxd/issues/165

### CODE REVIEW

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java [hunk 1]
```diff
@@ -91,6 +91,15 @@ private void register(RsResource rsResource) {
                 } else {
                     resource.setScopes(condition.getScopes());
                 }
+                //set creation and expiration timestamp
+                if (safeToInt(rsResource.getIat()) && safeToInt(rsResource.getExp())) {
```
Comment (yuriyz) [inline uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: it should be valid to set just `exp`. `iat` will be set by AS.

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java [hunk 2]
```diff
@@ -134,4 +143,11 @@ public ServiceProvider getServiceProvider() {
     public Map<Key, String> getIdMapCopy() {
         return Maps.newHashMap(idMap);
     }
+
+    public static boolean safeToInt(Integer input) {
```
Comment (yuriyz) [inline uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: isSafeToInt

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java [hunk 3]
```diff
@@ -91,6 +91,15 @@ private void register(RsResource rsResource) {
                 } else {
                     resource.setScopes(condition.getScopes());
                 }
+                //set creation and expiration timestamp
+                if (safeToInt(rsResource.getIat()) && safeToInt(rsResource.getExp())) {
+                    if(rsResource.getExp() > rsResource.getIat()){
+                        resource.setIat(rsResource.getIat());
+                        resource.setExp(rsResource.getExp());
+                    }else{
```
Comment (yuriyz) [inline uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: not formatted

**Review by yuriyz — CHANGES_REQUESTED** [review]

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java (hunk 1 above)
Comment (duttarnab) [reply uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: done

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java (hunk 2 above)
Comment (duttarnab) [reply uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: done

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java (hunk 3 above)
Comment (duttarnab) [reply uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: done

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java [hunk 4]
```diff
@@ -92,12 +92,13 @@ private void register(RsResource rsResource) {
                     resource.setScopes(condition.getScopes());
                 }
                 //set creation and expiration timestamp
-                if (safeToInt(rsResource.getIat()) && safeToInt(rsResource.getExp())) {
-                    if(rsResource.getExp() > rsResource.getIat()){
+                if (isSafeToInt(rsResource.getIat()) && isSafeToInt(rsResource.getExp())) {
```
Comment (yuriyz) [inline uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: if `iat` is null (not set) how would it go to line 101 ?

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java [hunk 5]
```diff
@@ -144,10 +145,7 @@ public ServiceProvider getServiceProvider() {
         return Maps.newHashMap(idMap);
     }
 
-    public static boolean safeToInt(Integer input) {
-        if (input != null && input > 0) {
-            return true;
-        }
-        return false;
+    public static boolean isSafeToInt(Integer input) {
+        return (input != null && input > 0);
```
Comment (yuriyz) [inline uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: parenthesis can be omitted

**Review by yuriyz — CHANGES_REQUESTED** [review]

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java (hunk 4 above)
Comment (duttarnab) [reply uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: Done. As date check already present in AS, so we removed `if(rsResource.getExp() > rsResource.getIat())` condition.

File: uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java (hunk 5 above)
Comment (duttarnab) [reply uma-rs-resteasy/src/main/java/org/gluu/oxd/rs/protect/resteasy/ResourceRegistrar.java]: done

**Review by yuriyz — APPROVED** [review]

**Pull request merged**